src/panels/lovelace/hui-root.js

```diff
@@ -119,7 +145,12 @@ class HUIRoot extends EventsMixin(PolymerElement) {
   }
 
   _handleViewSelected(ev) {
-    this._selectView(ev.detail.selected);
+    const index = ev.detail.selected;
+    if (index !== this._curView) {
+      const id = this.config.views[index].id || index;
+      this.navigate(`/lovelace/${id}`);
+    }
+    this._selectView(index);
```

Comment:
We shouldn't call this function either. It will cause the whole UI to be rebuild from scratch.